### ISSUE TITLE
💡 Feature Request - APIM missing DDOS recommendation check

### ISSUE THREAD

**Issue by cbezenco** [Azure/Azure-Proactive-Resiliency-Library-v2#543]
## Describe the solution you'd like

I recently went through a resiliency review where I found out my customer did not configure APIM DDOS protection. Their APIM was not configured with vnet-injected and therefore Azure DDOS protection could not be enabled on their APIM instances. In my customer case, there are 2 mission critical applications (including their e-commerce web site) leveraging this API server and this is clearly a potential resiliency weakness that should be raised through APRL to be consistent with other DDOS recommendation in the library.  
Reference : https://learn.microsoft.com/en-us/azure/api-management/protect-with-ddos-protection 

## Describe alternatives you've considered

NA, we need a rule that detect APIM configured without vnet injection and ensure customer there is no way to enable DDOS protection

## Additional context

Add any other context or screenshots about the feature request here. 📷

**Comment by ejhenry** [comment]
@cbezenco if I understand correctly, the ask is for a recommendation to always use vnet injected APIM. Is that accurate?

@aarthiem - please review

**Comment by cbezenco** [comment]
Yes. Based on the link shared above and the Note regarding potential DDOS vulnerability. 

APRL includes rules about exposure/risks linked to missing DDOS protection plan on Public IP, VNet and Azure Firewall. To be consistent, we should cover APIM potential DDOS vulnerability from my perspective.

**Comment by oZakari** [comment]
@ejhenry since these are more security focused than anything, should we actually pull these out of APRL?

**Comment by cbezenco** [comment]
My personal view, it should be consistent across all of APRL and we should explicitely mention that we are not covering this. Also If you do this you need to remove DDOS check on VNet and IP address. 

PS: I just went through an e-commerce site WARA and I can tell you that the lack of DDOS on APIM and their Kubernetes front-end load balancer was one of the major findings discussed during the WARA.
